### PR TITLE
chore: bump @floating-ui/dom

### PR DESCRIPTION
**Pull request by layershifter**
## New Behavior

Bumps `@floating-ui/dom` to the latest version, the change is also done in #33023.

I extracted it to a smaller PR to ensure that nothing will break.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

File: change/@fluentui-react-positioning-f474a804-4972-4d68-b0dc-f1d699251a90.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-positioning-f474a804-4972-4d68-b0dc-f1d699251a90.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33458&commitId=de9e18bde93b3c42cee20f946776fded7562828e&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Beak 25.chromium.png|2186| Changed|
|Callout.Right center.chromium.png|2396| Changed|
|Callout.Top left edge.chromium.png|2307| Changed|

</details>


  <details><summary><b>react-charting-AreaChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-AreaChart.Custom Accessibility.chromium.png|11| Changed|

</details>


  <details><summary><b>react-charting-LineChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Gaps.chromium.png|1| Changed|

</details>


  <details><summary><b>react-charting-VerticalBarChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-VerticalBarChart.Basic - Secondary Y Axis.chromium.png|4| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## 🕵 FluentUIV0 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v0"/>

File: change/@fluentui-react-positioning-f474a804-4972-4d68-b0dc-f1d699251a90.json (hunk 1 above)
Comment (fabricteam) [inline change/@fluentui-react-positioning-f474a804-4972-4d68-b0dc-f1d699251a90.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33458&commitId=de9e18bde93b3c42cee20f946776fded7562828e&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|5| Changed|
|Avatar Converged.badgeMask.chromium.png|11| Changed|

</details>


  <details><summary><b>Drawer</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay Dark Mode.chromium.png|984| Changed|
|Drawer.overlay drawer full.chromium.png|1138| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by mainframev — APPROVED** [review]

**Review by ling1726 — APPROVED** [review]

**Pull request merged**